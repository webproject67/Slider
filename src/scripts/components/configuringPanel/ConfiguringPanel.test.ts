import ConfiguringPanel from './ConfiguringPanel';

const state = {
  min: 0,
  max: 100,
  from: 0,
  fromPercent: 0,
  to: 100,
  toPercent: 100,
  step: 1,
  view: false,
  range: false,
  flag: true,
  progress: true,
  scale: true,
};

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new ConfiguringPanel(state).getElement()).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    state.range = true;
    expect(new ConfiguringPanel(state).getElement()).toMatchSnapshot();
  });
});

describe('handleInputChange', () => {
  const panel = new ConfiguringPanel(state).getElement();

  test('event input min', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputMinChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input max', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputMaxChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input from', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputFromChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input to', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputToChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input step', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputStepChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input view', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputViewChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input range', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputRangeChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input flag', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputFlagChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input progress', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputProgressChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });

  test('event input scale', () => {
    const input: HTMLInputElement | null = panel.querySelector(
      '.slider__handleInputScaleChange'
    );
    const change = new MouseEvent('change');
    if (input !== null) input.dispatchEvent(change);
  });
});
