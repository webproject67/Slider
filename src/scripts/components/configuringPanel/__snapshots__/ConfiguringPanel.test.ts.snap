// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
<div
  class="slider__labels"
>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Минимальное значение
    <input
      class="slider__handleInputMinChange"
      data-name="handleInputMinChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Максимальное значение
    <input
      class="slider__handleInputMaxChange"
      data-name="handleInputMaxChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
    style="display: none;"
  >
    От
    <input
      class="slider__handleInputFromChange"
      data-name="handleInputFromChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Текущее значение
    <input
      class="slider__handleInputToChange"
      data-name="handleInputToChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Шаг
    <input
      class="slider__handleInputStepChange"
      data-name="handleInputStepChange"
      type="number"
    />
  </label>
  <label
    class="slider__label"
  >
    Вертикальный
    <input
      class="slider__handleInputViewChange"
      data-name="handleInputViewChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Интервал
    <input
      class="slider__handleInputRangeChange"
      data-name="handleInputRangeChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Значение
    <input
      class="slider__handleInputFlagChange"
      data-name="handleInputFlagChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Шкала
    <input
      class="slider__handleInputScaleChange"
      data-name="handleInputScaleChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Прогресс
    <input
      class="slider__handleInputProgressChange"
      data-name="handleInputProgressChange"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`snapshot renders correctly element2 1`] = `
<div
  class="slider__labels"
>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Минимальное значение
    <input
      class="slider__handleInputMinChange"
      data-name="handleInputMinChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Максимальное значение
    <input
      class="slider__handleInputMaxChange"
      data-name="handleInputMaxChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    От
    <input
      class="slider__handleInputFromChange"
      data-name="handleInputFromChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    До
    <input
      class="slider__handleInputToChange"
      data-name="handleInputToChange"
      type="number"
    />
  </label>
  <label
    class="slider__label slider__label_state_displayed"
  >
    Шаг
    <input
      class="slider__handleInputStepChange"
      data-name="handleInputStepChange"
      type="number"
    />
  </label>
  <label
    class="slider__label"
  >
    Вертикальный
    <input
      class="slider__handleInputViewChange"
      data-name="handleInputViewChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Интервал
    <input
      class="slider__handleInputRangeChange"
      data-name="handleInputRangeChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Значение
    <input
      class="slider__handleInputFlagChange"
      data-name="handleInputFlagChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Шкала
    <input
      class="slider__handleInputScaleChange"
      data-name="handleInputScaleChange"
      type="checkbox"
    />
  </label>
  <label
    class="slider__label"
  >
    Прогресс
    <input
      class="slider__handleInputProgressChange"
      data-name="handleInputProgressChange"
      type="checkbox"
    />
  </label>
</div>
`;
